main/api.c

```diff
@@ -111,6 +125,22 @@ static void read_cb(evutil_socket_t sock, short what, void * /*priv*/) {
 		}
 	}
 
+	if (req.type == GR_MAIN_ENABLE_NOTIFICATIONS) {
+		out.status = 0;
+		out.len = 0;
+		gr_vec_add(notification_sockets, sock);
+		goto send;
+	} else if (req.type == GR_MAIN_DISABLE_NOTIFICATIONS) {
+		out.status = 0;
+		out.len = 0;
+		for (size_t i = 0; i < gr_vec_len(notification_sockets); i++) {
+			if (notification_sockets[i] == sock) {
+				gr_vec_del_swap(notification_sockets, i);
+				goto send;
+			}
+		}
```

Comment:
If GR_MAIN_DISABLE_NOTIFICATIONS is received while not subscribed yet, maybe you should return another error than ENOTSUP.